mantis-connectors/mantis-connector-iceberg/src/main/java/io/mantisrx/connector/iceberg/sink/committer/IcebergCommitter.java

```diff
@@ -16,36 +16,79 @@
 
 package io.mantisrx.connector.iceberg.sink.committer;
 
+import static io.mantisrx.connector.iceberg.sink.writer.DefaultIcebergWriter.minNullSafe;
+
+import io.mantisrx.connector.iceberg.sink.committer.config.CommitterConfig;
+import io.mantisrx.connector.iceberg.sink.writer.MantisDataFile;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.iceberg.AppendFiles;
 import org.apache.iceberg.DataFile;
 import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.UpdateProperties;
 
 
 /**
  * Commits {@link DataFile}s for Iceberg tables.
  *
  * This class uses Iceberg's Table API and only supports Table#append operations.
  */
+@Slf4j
 public class IcebergCommitter {
 
     private final Table table;
+    private final CommitterConfig config;
 
-    public IcebergCommitter(Table table) {
+    public IcebergCommitter(Table table, CommitterConfig committerConfig) {
         this.table = table;
+        this.config = committerConfig;
     }
 
     /**
      * Uses Iceberg's Table API to append DataFiles and commit metadata to Iceberg.
      *
      * @return the current snapshot of the table.
      */
-    public Map<String, Object> commit(List<DataFile> dataFiles) {
-        AppendFiles tableAppender = table.newAppend();
-        dataFiles.forEach(tableAppender::appendFile);
+    public Map<String, Object> commit(List<MantisDataFile> dataFiles) {
+        Transaction transaction = table.newTransaction();
+
+        AppendFiles tableAppender = transaction.newAppend();
+        dataFiles.stream().map(MantisDataFile::getDataFile).forEach(tableAppender::appendFile);
         tableAppender.commit();
+        log.info(
+            "Iceberg committer {}.{} appended {} data files to transaction",
+            config.getDatabase(),
+            config.getTable(),
+            dataFiles.size());
+
+        Long lowWatermark = getCurrentWatermark(transaction.table());
+        for (MantisDataFile flinkDataFile : dataFiles) {
+            lowWatermark = minNullSafe(lowWatermark, flinkDataFile.getLowWatermark());
+        }
+
+        if (lowWatermark != null) {
```

Comment:
I think we need to check the max of the current watermark and the new watermark.

Reply:
good point - done!